components/whitelist.tsx

```diff
@@ -0,0 +1,107 @@
+"use client";
+
+import { Button } from "@/components/ui/button";
+import { Input } from "@/components/ui/input";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { z } from "zod";
+import {
+  Form,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormControl,
+  FormMessage,
+} from "@/components/ui/form";
+import { useForm } from "react-hook-form";
+import { useState } from "react";
+
+export const whitelistSchema = z.object({
+  email: z.string().email({ message: "Email address is invalid." }),
+});
+
+export type WhitelistFormData = z.infer<typeof whitelistSchema>;
+
+export default function WhitelistPage() {
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+  const form = useForm<WhitelistFormData>({
+    resolver: zodResolver(whitelistSchema),
+    defaultValues: {
+      email: "",
+    },
+  });
+
+  const handleEmailSubmit = async (data: WhitelistFormData) => {
+    // TODO: endpoint on server /whitelist
```

Comment:
So we actually have a nextjs backend as well in this repository.

What we want is for this part to send request to our nextjs backend `app/api/whitelist` first as proxy, then you'd implement the `app/api/whitelist/route.ts` to actually call our python server's `whitelist` endpoint (not implemented yet).

You can take a look at the other `route.ts` files in `app/api` to have an idea.

This proxy makes our app a bit more secure and prevents leaking our server url